lightning/src/events/bump_transaction.rs

```diff
@@ -228,3 +239,359 @@ pub enum BumpTransactionEvent {
 		tx_lock_time: PackedLockTime,
 	},
 }
+
+pub struct Input {
+	pub outpoint: OutPoint,
+	pub witness_weight: u64,
+}
+
+#[derive(Clone)]
+pub struct Utxo {
+	pub outpoint: OutPoint,
+	pub output: TxOut,
+	pub witness_weight: u64,
+}
+
+pub struct CoinSelection {
+	confirmed_utxos: Vec<Utxo>,
+	fee_amount_sat: u64,
+	additional_tx_outputs: Vec<TxOut>,
+}
+
+pub trait CoinSelectionSource {
+	fn select_confirmed_utxos(
+		&self, must_spend: Vec<Input>, must_pay_to: Vec<TxOut>, target_feerate_sat_per_1000_weight: u32,
+	) -> Result<CoinSelection, ()>;
+	fn change_script(&self) -> Result<Script, ()>;
+	fn sign_confirmed_utxo(&self, tx: &Transaction, input: usize) -> Result<Witness, ()>;
+}
+
+pub trait WalletSource {
```

Comment:
so what's the reasoning behind the dichotomy `WalletSource` and `CoinSelectionSource`, like first one is for trusted coins (i.e controlled by the user) and the other one for third-party coins (i.e booked from a LSP services) or in batch with other channels?

Reply:
Okay need documentation.